File: modules/swagger-core/src/test/scala/converter/ATMTest.scala
package converter

import model._
import com.wordnik.swagger.converter._
import com.wordnik.swagger.model._
import com.wordnik.swagger.annotations._
import com.wordnik.swagger.converter._
import com.wordnik.swagger.core.util._
import com.wordnik.swagger.model._
import scala.reflect.BeanProperty
import scala.collection.mutable.LinkedHashMap
import org.junit.runner.RunWith
import org.scalatest.junit.JUnitRunner
import org.scalatest.FlatSpec
import org.scalatest.matchers.ShouldMatchers
import javax.xml.bind.annotation._
import converter.models.ATM

@RunWith(classOf[JUnitRunner])
class ATMTest extends FlatSpec with ShouldMatchers {
  it should "read a model with enums" in {
    val a = ModelConverters.readAll(classOf[ATM])
    JsonSerializer.asJson(a) should be ("""[{"id":"ATM","properties":{"currency":{"$ref":"Currency","enum":["USA","CANADA"]}}}]""")
  }
}

File: modules/swagger-core/src/test/scala/com/wordnik/swagger/jackson/ATMTest.java
package com.wordnik.swagger.jackson;

import com.wordnik.swagger.converter.ModelConverterContextImpl;
import com.wordnik.swagger.jackson.*;
import com.wordnik.swagger.models.*;

public class ATMTest extends SwaggerTestBase {
  static class ATM {
    private Currency currency;

    public void setCurrency(Currency currency) {
      this.currency = currency;
    }

    public Currency getCurrency() {
      return currency;
    }
  }

  public enum Currency { USA, CANADA }

  public void testATMModel() throws Exception {
    
	ModelResolver modelResolver = new ModelResolver(mapper());
	   ModelConverterContextImpl context = new ModelConverterContextImpl(modelResolver);
	
	Model model = context
       .resolve(ATM.class);
    assertNotNull(model);
    /*
    prettyPrint(model);
    */
  }
}

